.github/actions/prepare-ptaxsim/action.yaml

```diff
@@ -0,0 +1,77 @@
+name: Prepare PTAXSIM database
+description: Downloads and extracts the PTAXSIM database file from S3
+inputs:
+  PTAXSIM_DB_BASE_URL:
+    required: false
+    type: string
+    default: "s3://ccao-data-public-us-east-1/ptaxsim"
+  ASSUMED_ROLE:
+    description: "AWS role used for S3"
+outputs:
+  PTAXSIM_VERSION:
+    description: "PTAXSIM database version"
+    value: ${{ steps.version_db.outputs.PTAXSIM_VERSION }}
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Get database version
+      id: version_db
+      run: |
+        DESCRIPTION_PATH=$(echo "${{ github.workspace }}/DESCRIPTION" | sed 's/\\/\//g')
+        echo "PTAXSIM_VERSION=$(sed -n 's/.*Wants_DB_Version: \([0-9]*\.[0-9]*\.[0-9]\).*/\1/p' $DESCRIPTION_PATH)" >> $GITHUB_ENV
+        echo "PTAXSIM_VERSION=${{ env.PTAXSIM_VERSION }}" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - name: Restore database cache
+      uses: actions/cache/restore@v3.3.1
+      id: restore_db_cache
+      with:
+        path: ptaxsim.db.bz2
+        key: ${{ format('{0}-{1}', env.PTAXSIM_VERSION, hashFiles('DESCRIPTION')) }}
+        enableCrossOsArchive: true
+
+    - name: Configure AWS credentials
+      if: steps.restore_db_cache.outputs.cache-hit != 'true'
+      uses: aws-actions/configure-aws-credentials@v2
+      with:
+        role-to-assume: ${{ inputs.ASSUMED_ROLE }}
+        aws-region: us-east-1
+
+    - name: Fetch database file
+      id: fetch_db
+      if: steps.restore_db_cache.outputs.cache-hit != 'true'
+      run: |
+        aws s3 cp ${{ inputs.PTAXSIM_DB_BASE_URL }}/ptaxsim-${{ env.PTAXSIM_VERSION }}.db.bz2 ${{ github.workspace }}/ptaxsim.db.bz2 --quiet
+      shell: bash
+
+    - name: Unpack database (Linux)
+      if: runner.os == 'Linux'
+      run: |
+        sudo apt-get install -y pbzip2
+        pbzip2 -dk ptaxsim.db.bz2
+      shell: bash
+
+    - name: Unpack database (macOS)
+      if: runner.os == 'macOS'
+      run: |
+        brew install pbzip2
+        pbzip2 -dk ptaxsim.db.bz2
+      shell: bash
+
+    - name: Unpack database (Windows)
+      if: runner.os == 'Windows'
+      run: 7z x ptaxsim.db.bz2
+      shell: cmd
+
+    - name: Save database cache
+      uses: actions/cache/save@v3
+      id: save_db
+      if: steps.restore_db_cache.outputs.cache-hit != 'true'
+      with:
+        path: ptaxsim.db.bz2
+        key: ${{ format('{0}-{1}', env.PTAXSIM_VERSION, hashFiles('DESCRIPTION')) }}
+        enableCrossOsArchive: true
```

Comment:
**[Question, non-blocking]** I'm not quite an expert on GA caching, what's the benefit for this action of splitting `save` and `restore` into two steps instead of using the combined `actions/cache` action that handles both?